src/samplers/cpuidle/mod.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+pub(crate) mod statistics;
+
+use self::statistics::*;
+use crate::config::Config;
+use crate::samplers::{Common, Sampler};
+
+use failure::Error;
+use logger::*;
+use metrics::*;
+use time;
+
+use std::fs;
+
+pub struct CpuIdle<'a> {
+    common: Common<'a>,
+    cores: u64,
+}
+
+fn get_cpuidle_usage(cpu: u64, state: &Statistic) -> Result<u64, ()> {
+    let filename = format!("/sys/devices/system/cpu/cpu{}/{}/usage", cpu, state);
+    Ok(fs::read_to_string(filename)
```

Comment:
Probably not easy enough to discover, but I think we could maybe use the functions in `common::file` here. https://github.com/twitter/rezolus/blob/master/src/common/file.rs#L11

Reply:
Done.